test/Dialect/FIRRTL/errors.mlir

```diff
@@ -1261,6 +1261,16 @@ firrtl.circuit "PropertyDoubleDrive" {
   }
 }
 
+// -----
+// Check that you can't connect property types.
+
+firrtl.circuit "PropertyConnect" {
+  firrtl.module @PropertyConnect(out %out: !firrtl.string) {
+    %0 = firrtl.string "hello"
+    // expected-error @below {{must be a passive base type}}
+    firrtl.strictconnect %out, %0 : !firrtl.string
```

Comment:
Error message isn't awesome here. However, this is locking in existing errors. So... 👍

Reply:
Oh, good call, was splitting from other work so didn't think too hard :innocent: .  Thanks for pointing this out, regardless!  You're right this is just locking in existing behavior, but on this.....

I wonder how we could generate better error messages for violations of ODS constraints?
We could drop them in some places (darn) in favor of manual verifiers where we control the diagnostics,
but they're nice to have in ODS and are somewhat necessary/significant directly (MLIR detects if you're using a constraint derived from X  to know to generate/reason about Y in a certain way, so on (e.g., SameTypeOperands, TypesMatchWith).

We could in some places decompose them into compositions of more specific errors (not a property type, so on)... :thinking:

Reply:
Looks like we already produce nice diagnostic for this if encountered during parsing, so unlikely user will see this, phew! :+1: .